repo/fsrepo/defaultds.go

```diff
@@ -31,16 +31,8 @@ func openDefaultDatastore(r *FSRepo) (repo.Datastore, error) {
 		return nil, fmt.Errorf("unable to open leveldb datastore: %v", err)
 	}
 
-	// 4TB of 256kB objects ~=17M objects, splitting that 256-way
-	// leads to ~66k objects per dir, splitting 256*256-way leads to
-	// only 256.
-	//
-	// The keys seen by the block store have predictable prefixes,
-	// including "/" from datastore.Key and 2 bytes from multihash. To
-	// reach a uniform 256-way split, we need approximately 4 bytes of
-	// prefix.
 	syncfs := !r.config.Datastore.NoSync
-	blocksDS, err := flatfs.New(path.Join(r.path, flatfsDirectory), 4, syncfs)
+	blocksDS, err := flatfs.New(path.Join(r.path, flatfsDirectory), 7, syncfs)
```

Comment:
@whyrusleeping.  Our messages crossed.  It looks like the prefix was _not_ included in the prefix length before ipfs/go-datastore#39 but now it is (see my comments on the pull request).  I will leave it up to you to determine the best value.  (To avoid too many directories I believe a smaller value would be better.)